airflow/executors/kubernetes_executor.py

```diff
@@ -772,7 +773,15 @@ def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[Task
         kube_client: client.CoreV1Api = self.kube_client
         for scheduler_job_id in scheduler_job_ids:
             scheduler_job_id = pod_generator.make_safe_label_value(str(scheduler_job_id))
-            query_kwargs = {"label_selector": f"airflow-worker={scheduler_job_id}"}
+            # We will look for any pods owned by the no-longer-running scheduler,
+            # but will exclude only successful pods, as those TIs will have a terminal state
+            # and not be up for adoption!
+            # Those workers that failed, however, are okay to adopt here as their TI will
+            # still be in queued.
+            query_kwargs = {
+                "field_selector": "status.phase!=Succeeded",
+                "label_selector": f"kubernetes_executor=True,airflow-worker={scheduler_job_id}",
```

Comment:
i'll bet you there's a selector IN but... don't care :)

Reply:
I was going to say "actually, no", but turns out there is for [labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) but not for [fields](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#supported-operators).

Future work!

Reply:
🙃